config/config.go

```diff
@@ -40,16 +42,94 @@ func (e MissingFieldError) Error() string {
 	return fmt.Sprintf("field missing or unset: %s", e.Field)
 }
 
+// DuplicateNameError is an error that includes the duplicate name.
+type DuplicateNameError struct {
+	Name string
+}
+
+func (e DuplicateNameError) Error() string {
+	return fmt.Sprintf("found duplicate name: %s", e.Name)
+}
+
+// normalize lowercases, and removes all non-alphanumeric characters from a string.
+func normalize(s string) string {
+	regex := regexp.MustCompile("[^a-zA-Z0-9]+")
+	s = regex.ReplaceAllString(s, "")
+	s = strings.ToLower(s)
+	return s
+}
+
+// validateUnique checks that a list has no duplicate entries.
+func validateUnique(items []string) error {
+	mErr := &multierror.Error{}
+	set := map[string]bool{}
+	for _, s := range items {
+		_, ok := set[s]
+		if ok {
+			mErr = multierror.Append(mErr, DuplicateNameError{s})
+		} else {
+			set[s] = true
+		}
+	}
+	return mErr.ErrorOrNil()
+}
+
 // Validate checks that a configuration is well-formed, having test groups and dashboards set.
+// Also checks each group of entities (TestGroup, DashboardTab, Dashboard, DashboardGroup) for
+// unique names and valid options.
 func Validate(c configpb.Configuration) error {
+	mErr := &multierror.Error{}
+
 	if len(c.TestGroups) == 0 {
 		return MissingFieldError{"TestGroups"}
 	}
 	if len(c.Dashboards) == 0 {
 		return MissingFieldError{"Dashboards"}
 	}
 
-	return nil
+	tgNames := []string{}
+	for _, tg := range c.TestGroups {
+		tgNames = append(tgNames, normalize(tg.GetName()))
+	}
+	err := validateUnique(tgNames)
```

Comment:
Ex. `Test Group names must be unique after being normalized.`

Reply:
Done